lib/simplecov/result_adapter.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module SimpleCov
+  #
+  # Resposible of adapting the format of the coverage result weather it's default or with statistics
+  #
+  #
+  class ResultAdapter
+    attr_reader :result
+
+    def initialize(result)
+      @result = result
+    end
+
+    def self.call(*args)
+      new(*args).adapt
+    end
+
+    def adapt
+      return result unless result
```

Comment:
Maybe we can replace it with `return unless result`?

Reply:
Yes , I agree.